client/src/test/scala/org/http4s/client/JettyScaffold.scala

```diff
@@ -6,40 +6,31 @@ import javax.servlet.http.HttpServlet
 import org.eclipse.jetty.server.{Server => JServer, ServerConnector}
 import org.eclipse.jetty.servlet.{ServletHolder, ServletContextHandler}
 import org.http4s.Http4sSpec
+import org.specs2.mutable.SpecificationLike
+import org.specs2.specification.BeforeAfterAll
 import org.specs2.specification.core.Fragments
 
 import scala.concurrent.duration._
 
-
-abstract class JettyScaffold(name: String) extends Http4sSpec {
+trait JettyScaffold extends SpecificationLike {
 
   private val server = new JServer()
+  var address: InetSocketAddress = null
 
-  def testServlet(): HttpServlet
-
-  protected def runAllTests(): Fragments
-
-  // Start the tests
-  name >> {
-    step(startup()) ^
-      runAllTests()   ^
-      step(cleanup())
-  }
-
-  def startup() = {}
+  def testServlet: HttpServlet
 
-  def cleanup() = {
-    server.stop()
+  override def map(fs: => Fragments) = {
+    step(startServer()) ^ fs ^ step(server.stop())
   }
 
   protected def timeout: Duration = 10.seconds
 
-  protected def initializeServer(): InetSocketAddress = {
-    val address = new InetSocketAddress(InetAddress.getLocalHost(), JettyScaffold.getNextPort())
+  private def startServer(): InetSocketAddress = {
```

Comment:
There is already logging from jetty and our clients if we turn on `LOGBACK_ROOT_LEVEL=debug`.  We have that off in Travis, because there is so much debug logging, it was hard to determine causes of failure.